core/tests/puppeteer-acceptance-tests/user-utilities/curriculum-admin-utils.ts

```diff
@@ -0,0 +1,412 @@
+const topicUrlFragmentField = 'input.e2e-test-new-topic-url-fragment-field';
+const topicWebFragmentField = 'input.e2e-test-new-page-title-fragm-field';
+const topicDescriptionField = 'textarea.e2e-test-new-topic-description-field';
+const createTopicButton = 'button.e2e-test-confirm-topic-creation-button';
+const saveTopicButton = 'button.e2e-test-save-topic-button';
+
+const addSubTopicButton = 'button.e2e-test-add-subtopic-button';
+const subTopicTitleField = 'input.e2e-test-new-subtopic-title-field';
+const subTopicUrlFragmentField =
+  'input.e2e-test-new-subtopic-url-fragment-field';
+const subTopicDescriptionEditorToggle = 'div.e2e-test-show-schema-editor';
+const createSubTopicButton = '.e2e-test-confirm-subtopic-creation-button';
+
+const skillDescriptionField = 'input.e2e-test-new-skill-description-field';
+const skillReviewMaterialHeader = 'div.e2e-test-open-concept-card';
+const addSkillButton = 'button.e2e-test-add-skill-button';
+const confirmSkillCreationButton =
+  'button.e2e-test-confirm-skill-creation-button';
+
+const addStoryButton = 'button.e2e-test-create-story-button';
+const storyTitleField = 'input.e2e-test-new-story-title-field';
+const storyDescriptionField = 'textarea.e2e-test-new-story-description-field';
+const storyUrlFragmentField = 'input.e2e-test-new-story-url-fragment-field';
+const createStoryButton = 'button.e2e-test-confirm-story-creation-button';
+const saveStoryButton = 'button.e2e-test-save-story-button';
+const saveStoryMessageInput = 'textarea.e2e-test-commit-message-input';
+const publishStoryButton = 'button.e2e-test-publish-story-button';
+
+const addChapterButton = 'button.e2e-test-add-chapter-button';
+const chapterTitleField = 'input.e2e-test-new-chapter-title-field';
+const chapterExplorationIdField = 'input.e2e-test-chapter-exploration-input';
+const createChapterButton = 'button.e2e-test-confirm-chapter-creation-button';
+
+export class CurriculumAdmin extends BaseUser {
+  /**
+   * Function for navigating to the topic and skills dashboard page.
+   */
+  async navigateToTopicAndSkillsDashboardPage(): Promise<void> {
+    await this.goto(topicAndSkillsDashboardUrl);
+  }
+
+  /**
+   * Function for creating a skill in the topics and skills dashboard.
+   */
+  async createSkill(): Promise<void> {
+    await this.openTopicEditor();
+    await this.clickOn(addSkillButton);
+    await this.type(skillDescriptionField, 'Test Skill 1');
+    await this.clickOn(skillReviewMaterialHeader);
+    await this.clickOn(richTextAreaField);
+    await this.type(
+      richTextAreaField,
+      'This is a test skill for curriculum admin.'
+    );
+    await this.page.waitForSelector(
+      `${confirmSkillCreationButton}:not([disabled])`
+    );
+    await this.page.waitForTimeout(500);
+    await this.clickOn(confirmSkillCreationButton);
+    await this.page.bringToFront();
+  }
+
+  /**
+   * Function for creating a question in the skill editor page.
+   */
+  async createQuestion(): Promise<void> {
```

Comment:
Many of the other functions start by going to a particular page -- shouldn't this one do so too, or at least assert that the current page is the correct one?

Ditto for other functions -- this seems important for this user because they can do things in quite a lot of places.

Reply:
You're right, it currently has to be in flow and has to be called write after creating an exploration for it to work. Updated it so that it works regardless of the order.

Other functions all are independent aside from creating chapters and publishing the story, we do not want to refresh the page in the middle of story creation.

Reply:
OK, sounds good, but one note: if you expect function A to be called before function B then either combine those into one function and don't expose the inner ones, or have an assertion at the start of B that the URL points to the correct page.